internal/certprovider.go

```diff
@@ -1,77 +1,160 @@
 package internal
 
 import (
+	"archive/zip"
+	"bytes"
+	"crypto/sha256"
 	"crypto/x509"
 	_ "embed"
+	"encoding/hex"
 	"encoding/pem"
 	"fmt"
+	"io"
+	"net/http"
+	"strings"
 )
 
-// asserts/aws_nitro_enclaves.crt contains the PEM encoded roots for verifying Nitro
-//	Enclave attestation signatures. You can download them from
-//	https://aws-nitro-enclaves.amazonaws.com/AWS_NitroEnclaves_Root-G1.zip
-//	It's recommended you calculate the SHA256 sum of this string and match
-//	it to the one supplied in the AWS documentation
+// AWS_NitroEnclaves_Root-G1.zip file and its digest are advertised on
 //	https://docs.aws.amazon.com/enclaves/latest/user/verify-root.html
 
-//go:embed assets/aws_nitro_enclaves.crt
-var NitroCertPEM []byte
+const AWSNitroEnclavesRootURL = "https://aws-nitro-enclaves.amazonaws.com/AWS_NitroEnclaves_Root-G1.zip"
 
-//go:embed assets/selfsigned_cert.der
-var selfSignedCertDER []byte
+//go:embed assets/AWS_NitroEnclaves_Root-G1.zip
+var AWSNitroEnclavesRootZip []byte
 
-type NitroCertProvider struct {
-	certs *x509.CertPool
-}
+//go:embed assets/AWS_NitroEnclaves_Root-G1.sha256.hex
+var AWSNitroEnclavesRootSHA256Hex string
 
-func NewNitroCertProvider() *NitroCertProvider {
-	return &NitroCertProvider{}
-}
+type UnzipAWSRootCertsFunc func(zipBytes []byte) (pemBytes []byte, err error)
 
-func (cp *NitroCertProvider) Roots() (*x509.CertPool, error) {
-	if cp.certs == nil {
-		certs, err := cp.RootsWithCerts(NitroCertPEM)
-		if err != nil {
-			return nil, err
-		}
-		cp.certs = certs
+func UnzipAWSRootCerts(zipBytes []byte) ([]byte, error) {
+	zipReader, err := zip.NewReader(
+		bytes.NewReader(zipBytes),
+		int64(len(zipBytes)),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating zip reader: %w", err)
 	}
 
-	return cp.certs, nil
+	if len(zipReader.File) != 1 {
+		return nil, fmt.Errorf("unexpected file count: %d", len(zipReader.File))
+	}
+	if zipReader.File[0].Name != "root.pem" {
+		return nil, fmt.Errorf("unexpected file name: %s", zipReader.File[0].Name)
+	}
+
+	pemFile, err := zipReader.File[0].Open()
+	if err != nil {
+		return nil, fmt.Errorf("opening zip file: %w", err)
+	}
+
+	pemBytes, err := io.ReadAll(pemFile)
+	if err != nil {
+		return nil, fmt.Errorf("reading pem file: %w", err)
+	}
+
+	return pemBytes, nil
+}
+
+func NewEmbeddedRootCertZipReader() io.ReadCloser {
+	return io.NopCloser(bytes.NewReader(AWSNitroEnclavesRootZip))
 }
 
-func (_ *NitroCertProvider) RootsWithCerts(
-	pemCerts []byte,
+func NewFetchingRootCertZipReader() (io.ReadCloser, error) {
+	return NewFetchingRootCertZipReaderWithClient(http.DefaultClient)
+}
```

Comment:
Create an `io.ReadCloser` to either read the embedded zip file, or fetch one from the internet.